enhancements/network/user-defined-network-segmentation.md

```diff
@@ -0,0 +1,947 @@
+to work and reports 200 OK to Kubelet, thus rendering the pod serving in the endpoint slice, but unable to function.
+Although this is an unlikely scenario, it is good to document.
+
+### Network Policy
+
+Configuring network policies that allow traffic between user-defined network namespaces will not be invalidated or
+ignored by OVN-Kubernetes. If a user configures a network policy that would have no effect, (like stopping namespace A
+from talking to B on different networks) then OVN-Kubernetes will still configure this policy in OVN. These policies
+will also have no effect between namespaces that are assigned to different networks as there will be no network
+connectivity between the namespaces. However, there may still be a desire to configure network policy between namespaces
+that share the same network, as well as for other reasons like ingress/egress control. Additionally, policies that
+block/allow cluster egress or ingress traffic will still be enforced.
+
+### API Extensions
+
+The main API extension here will be a namespace scoped network CRD as well a cluster scoped network CRD. These CRDs
+will be registered by Cluster Network Operator (CNO). See the [CRDs for Managing Networks](#crds-for-managing-networks)
+section for more information on how the CRD will work. There will be a finalizer on the CRDs, so that upon deletion
+OVN-Kubernetes can validate that there are no pods still using this network. If there are pods still attached to this
+network, the network will not be removed.
+
+### Workflow Description
+
+#### Tenant Use Case
+
+As a tenant I want to ensure when I create pods in my namespace their network traffic is isolated from other tenants on
+the cluster. In order to ensure this, I first create a network CRD that is namespace scoped and indicate:
+
+ - Type of network (Layer 3 or Layer 2)
+ - IP addressing scheme I wish to use (optional)
+ - Indicate this network will be the primary network
+
+After creating this CRD, I can check the status of the CRD to ensure it is actively being used as the primary network
+for my namespace by OVN-Kubernetes. Once verified, I can now create pods and they will be in their own isolated SDN.
+
+#### Admin Use Case
+
+As an admin, I have a customer who has multiple namespaces and wants to connect them all to the same private network. In
+order to accomplish this, I first create an admin network CRD that is cluster scoped and indicate:
+
+- Type of network (Layer 3 or Layer 2)
+- IP addressing scheme I wish to use (optional)
+- Indicate this network will be the primary network
+- Selector to decide which namespaces may connect to this network. May use the ```kubernetes.io/metadata.name``` label to 
+guarantee uniqueness and eliminates the ability to falsify access.
+
+After creating the CRD, check the status to ensure OVN-Kubernetes has accepted this network to serve the namespaces
```

Comment:
how do we map admin scope CRD with a NAD object ? I guess NAD is namespace scoped object.

Reply:
The NAD could reside in some  registered namespace that allows cross-namespace networking, similar to default namespace.
Such namespace should probably registered by CNO https://github.com/k8snetworkplumbingwg/multus-cni/blob/master/docs/configuration.md#allow-specific-namespaces-to-be-used-across-namespaces-when-using-namespace-isolation

Reply:
Here is an option, the admin scope CRD represent cross-namespace network, each corresponding NAD (on each relevant namesapce) could use the same network-name (CNI conf `name`).
That way we could have multiple NADs, on different namespaces, pointing to the same underlying OVN network.